src/coreclr/src/vm/nativeimage.cpp

```diff
@@ -53,7 +53,7 @@ void NativeImage::Initialize(READYTORUN_HEADER *pHeader, LoaderAllocator *pLoade
 
     m_pReadyToRunInfo = new ReadyToRunInfo(/*pModule*/ NULL, m_pImageLayout, pHeader, /*compositeImage*/ NULL, pamTracker);
     m_pComponentAssemblies = m_pReadyToRunInfo->FindSection(ReadyToRunSectionType::ComponentAssemblies);
```

Comment:
Related: 


We should either only have `IMAGE_DATA_DIRECTORY *m_pComponentAssemblies;` field and always fetch the count for it (ie do not cache it in the field);

or we should have both the pointer resolved in the field, like:
```
    PTR_READYTORUN_COMPONENT_ASSEMBLIES_ENTRY m_pComponentAssemblies;
    DWORD                                     m_nComponentAssemblies;
```